src/storage/storage_structure/overflow_file.cpp

```diff
@@ -163,11 +163,16 @@ OverflowFile::OverflowFile(const DBFileIDAndName& dbFileIdAndName, BufferManager
     : bufferManager{bufferManager}, wal{wal}, headerChanged{false} {
     auto overflowFileIDAndName = constructDBFileIDAndName(dbFileIdAndName);
     dbFileID = overflowFileIDAndName.dbFileID;
-    fileHandle = bufferManager->getBMFileHandle(overflowFileIDAndName.fName,
-        readOnly ? FileHandle::O_PERSISTENT_FILE_READ_ONLY :
-                   FileHandle::O_PERSISTENT_FILE_CREATE_NOT_EXISTS,
-        BMFileHandle::FileVersionedType::VERSIONED_FILE, vfs, context);
-    if (fileHandle->getNumPages() > HEADER_PAGE_IDX) {
+    if (vfs) {
+        KU_ASSERT(bufferManager && context && wal);
+        fileHandle = bufferManager->getBMFileHandle(overflowFileIDAndName.fName,
+            readOnly ? FileHandle::O_PERSISTENT_FILE_READ_ONLY :
+                       FileHandle::O_PERSISTENT_FILE_CREATE_NOT_EXISTS,
+            BMFileHandle::FileVersionedType::VERSIONED_FILE, vfs, context);
+    } else {
+        fileHandle = nullptr;
```

Comment:
Since the fileHandle now may be null, I think we should probably either assert that it's non-null in functions that use it, or maybe split up the class into an in-memory version and one that supports disk operations.